inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -2237,6 +2228,15 @@ void TransposeWeightsFromNCHWToNHWCPass::run() {
                 }
                 auto weightable = dynamic_cast<WeightableLayer*>(l.get());
                 IE_ASSERT(weightable != nullptr);
+                size_t totalElements = 0;
+                auto totalWeights = weightable->_weights->size();
+                for (auto && transpositionInfoPart : transpositionInfo) {
+                        totalElements += transpositionInfoPart.num_transpose_rows * transpositionInfoPart.num_transpose_columns;
+                }
+                if (totalElements != totalWeights) {
+                        THROW_GNA_EXCEPTION << l->name << " weights columns from transposition info (" << totalElements
+                                            << ") don't match input dimensions (" << totalWeights << ")";
+                    }
```

Comment:
since this code is similar with one for FullyConnected layer I'd create a lambda function (something like transpInfoMatchWeightsSize) and call it in all required places

Reply:
ok, that makes sense